core/sail/nativerdf/src/main/java/org/eclipse/rdf4j/sail/nativerdf/SequentialRecordCache.java

```diff
@@ -55,36 +89,78 @@ public SequentialRecordCache(File cacheDir, int recordSize) throws IOException {
 	public SequentialRecordCache(File cacheDir, int recordSize, long maxRecords) throws IOException {
 		super(maxRecords);
 		this.recordSize = recordSize;
+		this.cacheDir = cacheDir;
+	}
 
-		File cacheFile = File.createTempFile("txncache", ".dat", cacheDir);
-		nioFile = new NioFile(cacheFile);
+	private void init() throws IOException {
```

Comment:
Don't really need to use lazy initialization, but it was fun to play with VarHandle a bit. Going to revert back I think and remove the VarHandle code.